cli/maintainer/src/maintenance.rs

```diff
@@ -1015,8 +1015,15 @@ impl SolidoState {
             // of Lamports, we don't bother withdrawing. We try to do this
             // so we don't pay more for fees than the amount that we'll
             // withdraw. Or if we have stake to remove from unstake accounts.
-            if expected_difference_stake > SolidoState::MINIMUM_WITHDRAW_AMOUNT
-                || removed_unstake > Lamports(0)
+            // If we're dealing with an inactive validator, we try to withraw
+            // all inactive stake.
+            let minimum_withdraw_amount = if validator.entry.active {
+                SolidoState::MINIMUM_WITHDRAW_AMOUNT
+            } else {
+                Lamports(0)
+            };
+
+            if expected_difference_stake > minimum_withdraw_amount || removed_unstake > Lamports(0)
```

Comment:
this will not pass if validator is inactive
```rust
if expected_difference_stake > minimum_withdraw_amount
```
`minimum_withdraw_amount` would be 0 also as `expected_difference_stake`, so `0 > 0` is false.
Take a look [here](https://github.com/lidofinance/solido/commit/578d650dd3dcf3426afb1896e87479cf2f8ba848)
We should recompute effective stake balance after stake merging

Reply:
Why would `expected_difference_stake` be zero?

> if validator is inactive

Let’s distinguish two things. There is the `.active` field on `Validator`. And there are the stake accounts. The case that was failing happens when the `active` field is `false`, but there is active stake in the stake accounts, which the maintainer tries to deactivate. So in this case, `current_stake_balance` definitely greater than zero. `effective_stake_balance` is the difference between the recorded balance in stake accounts (which will be greater than zero) and unstake accounts (which will still be zero). If there is excess stake to withdraw, then `expected_difference_stake` is going to be greater than zero.

Am I missing something?

Reply:
Did you test this commit? I did. After stakes are merged the effective stake balance does not change by itself. It is changed only in WithdrawInactiveStake, but this is what we try to call here. Merging two stakes does not account additional rent exempt amount